<a href="https://colab.research.google.com/github/jdansb/Econophysics/blob/main/files/contsocial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  ## Basic Concepts


The double-entry bookkeeping principle states that for every debit entry there must always correspond another credit entry of the same value. Internal balance refers to the requirement of equality between debit and credit values in each account, while external balance implies the need for balance across all accounts in the system.

Once accounting is chosen as the instrument for the macroscopic measurement of economic movement, it follows that the economy of an entire country can be seen as that of a single large firm: the results of its operation over a given period are presented by the accounts that make up the system of national accounts.

What is conventionally called “social accounting” is not limited to the system of national accounts, but also includes other key components such as the balance of payments, the monetary system accounts, and social indicators such as income distribution and the human development index. For this reason, the analogy should be restricted solely to its form — accounts, balance sheets, and bookkeeping entries. Its substance and objectives are entirely different.

Since the accounting we are referring to is social in nature, all the statistical “effort” must serve to give civil society a clearer idea of the direction of a country, and thus allow it to intervene in these directions when necessary.

In the economic system in which we live, everything can be evaluated monetarily. Thus, the immense range of different goods and services that an economy is capable of producing can be transformed into a thing of the same substance, namely money. This is what makes it possible to measure aggregates such as national product and national income.

One of the fundamental notions of social accounting is that of identity. But here identity has a particular meaning: it is an accounting identity, and therefore the symbol $\equiv$ is usually used instead of the equality sign $=$. If we take exchange as an example, it implies two operations that are the inverse of one another — the buyer exchanges \$10 for a shirt and the seller exchanges a shirt for \$10 — but which, from an analytical point of view, form an identity, since one cannot exist without the other. Examples: just as a purchase cannot occur without a sale, neither can production exist without simultaneously constituting an expenditure and a generation of income. It is from the identity
$$
\text{product} \equiv \text{expenditure} \equiv \text{income}
$$
that the circular flow of income is derived. Similarly, saving necessarily implies investment, and investment cannot be understood without also considering it, in turn, as saving.

To arrive at the aggregate product of the economy, it is necessary to deduct from the gross value of production the value of intermediate consumption, that is, what was used as input in production. Here, by final good we mean a good that is sold to final consumers, that is, a good that will not be used as an input. By definition, every final good must have its value included in the calculation of output, but not every good whose value enters the calculation of output is a final good by nature. This is because those products that have not yet been consumed, and are not considered final products (that is, were produced to be used as inputs), are also included in the final product. In other words, the aggregate product of the economy is the combination of products sold to final consumers and products not consumed in the period. A final consumer is the economic agent who acquires a good or service that will not be used for resale or as an input in a new production process.

This approach is known as the expenditure approach and evaluates the product of an economy by considering the sum of the values of all goods and services produced in the period that were not destroyed or absorbed as inputs in the production of other goods and services. However, there are still two other ways to approach the issue. From the product (output) perspective, the evaluation of the total product of the economy consists of considering the value actually added by the production process in each productive unit. That is, if we look firm by firm, the value added is given by the difference between the value of its production and the value of the production it acquired as inputs. This is effectively the contribution of each firm to the formation of the total product of the economy.

Both results must be identical. But there is still another way to approach the issue, namely through the income perspective. We must begin by understanding what is called the production process: let us initially assume that there are only two factors of production: labor and another which we will generically call capital — concretely, this includes not only machines and other equipment, but the entire set of elements that constitute the objective conditions without which the production process cannot take place. It is therefore between capital and labor that the product generated by the economy must be distributed, since their participation in the production process is what ensured the creation of this product. It is also possible to include land as a third factor, but this does not substantially change the discussion. The remuneration of the labor factor is called wages, and the remuneration of the capital factor is called profit. Land, in turn, would correspond to rent.

Thus, over a given period of time, the remuneration of both factors, taken jointly, must equal in value the product obtained by the economy in that same period, since these remunerations are nothing more than the distribution of the product. The remunerations paid constitute what we call income. It is not difficult to see that, in this way, the identity
$$
\text{product} \equiv \text{income}
$$
is established. Therefore, the product generated by an economy in a given period of time is equal to the income generated in that same period.

Some additional remarks are that here, unlike in Marxism, money and currency are presented as synonyms, as are value and price. Moreover, society is divided into two groups in a different way. It reads: “The members that constitute society appear twice in the process of its material reproduction and play two distinct roles: at one moment, they are producers; at another, they appear as consumers of what has been produced.” In other words, both the worker and the capitalist are simultaneously producers and consumers. This also becomes evident when we see the factors of production—“labor” and “capital”—side by side, each being responsible for producing “value” equally. That is, in the production process, the worker contributes labor, and the capitalist contributes capital, both in parity. This also appears in the next emphasis given by the book: in addition to playing the role of consumers, households also hold the condition of owners of the factors of production, and it is in this capacity that they secure their access to the goods and services produced. Here, by “ownership of the factors of production,” we understand that workers only possess labor power to sell.

We thus have that the output approach considers the activity of individuals as producers, that is, the activity of productive units or firms. Finally, the income approach analyzes individuals in their condition as owners of factors of production. Transactions occur between households and firms and involve reciprocally determined flows of concrete goods and services, on the one hand, and money, on the other. Following the book, we have this flow:

1. Households supply firms with the factors of production they own and, in return, receive from firms an income, that is, a monetary remuneration;

2. Firms combine these factors in a process called the production process and obtain, as a result, a set of goods and services;

3. With the income received in exchange for the use, in production, of the factors they own, households purchase from firms the goods and services produced by them;

4. Households consume the goods and services;

Observe how this flow places “households” of workers and households of capitalists on equal footing. Income (wages or profits) is seen as a “reward” for their contribution to the production process. Concluding the chapter, we must clearly understand that the flow is that of income and not of product: the money that remunerates the factors of production is the same that returns to firms in the purchase of final goods and services. This does not occur with other goods.

## Basic Structure of National Accounts

The greatest simplifications are typically to consider a closed economy without government; we will work with this example.

Indeed, everything that is produced in a given period but is not consumed in that same period—meaning that it gives rise to consumption in the future—has a name: it is called investment. Investment is divided into fixed capital formation and changes in inventories. Investment is usually split into changes in inventories, which group goods whose future consumption occurs all at once, and gross fixed capital formation, which comprises goods that do not disappear after a single use and enable production over a given period of time.

The depreciation of fixed capital is called depreciation: fixed capital goods also wear out over time and through use, so that, at the end of a given period, their value will have been entirely absorbed by the flow of production of goods that occurred during that period. To obtain the net product of an economy in a given period, it is necessary to deduct from the gross product the portion destined for the replacement of the economy’s capital stock, that is, depreciation.

|       | **Debit**          |       | **Credit**               |
|-------|-------------------|-------|--------------------------|
| **A** | Net Product       | **C** | Personal Consumption     |
| **B** | Depreciation      | **D** | Change in Inventories    |
|       |                   | **E** | Gross Fixed Capital Formation |
|       | **Gross Product** |       | **Gross Expenditure**    |

With respect to remuneration, in addition to wages and profits, we must introduce two other categories of remuneration: rents (which remunerate owners of real estate in general) and interest (which remunerates owners of monetary capital). The only additional caution that must be taken is to avoid double counting, which may occur if we include in these items not only rents and interest paid to households, but also those paid to firms. The latter must not be considered because, as revenues, they are already included in the profit and loss statements of firms. The exception is the financial sector, but this (maybe) will be discussed later.

| **Debit**          |                                   |   | **Credit**                     |
|--------------------|-----------------------------------|---|--------------------------------|
| $\boldsymbol{a}_{1}$ | wages                             | C | Personal Consumption           |
| $\boldsymbol{a}_{2}$ | profits                          | D | Change in inventories         |
| $\boldsymbol{a}_{3}$ | rents                            | E | Gross Fixed Capital Formation |
| $\boldsymbol{a}_{4}$ | interest                         |   |                                |
| **A**              | Net product: $A=\sum a_{i}$      |   |                                |
| **B**              | Depreciation                     |   |                                |
|                    | **Gross Product**                    |   | **Gross Expenditure**             |

The production account shows the identity between income and expenditure, while the appropriation account shows how households allocate the income received from the lending of their factors of production to firms.

The capital account, in turn, shows the identity $\text{investment} \equiv \text{savings},$ which is nothing more than an alternative way of representing the identity $\text{product} \equiv \text{income} \equiv \text{expenditure}.$

### Opening the economy

So far we have worked with the simplest case, which is the one I should focus my research on, but it is interesting to relax the assumptions a bit. For that, we need to introduce a few additional concepts.

• Trade balance: the trade of goods with the rest of the world.

• Services and income balance: trade in services and factors of production (incomes such as profits and interest).

We now make a distinction between domestic product and national product. There is no single convention, but according to the SNA 93 we use the attribute “domestic” for “product” (domestic product), and “national” for “income” (national income). The idea is that we are usually interested, on the one hand, in GDP (gross domestic product), that is, the total product produced within the territory of the country, regardless of the origin of the factors of production responsible for it. On the other hand, we consider GNI (gross national income), which is the value added generated by factors of production owned by residents, regardless of the territory where this value is generated.

To obtain the national product of an economy, one must subtract from its domestic product the net income sent abroad or, if applicable, add to its domestic product the net income received from abroad. In most cases, developed countries are net exporters of capital and therefore receive net income from abroad, while the opposite occurs in less developed countries. We now proceed to the third version of the product account:

|                          | **Debit**                               |       | **Credit**                          |
|--------------------------|-----------------------------------------|-------|-------------------------------------|
| **I**                    | Imports of goods and services           | **G** | Exports of goods and services       |
| **J-H**                  | Net income sent [+] abroad              | **C** | Personal consumption                |
| **$\boldsymbol{a}_{1}$**   | wages                                   | **D** | Change in inventories               |
| **$\boldsymbol{a}_{2}$**   | profits                                 | **E** | Gross fixed capital formation       |
| **$\boldsymbol{a}_{3}$**   | rents                                  |       |                                     |
| **$\boldsymbol{a}_{4}$**   | interest                               |       |                                     |
| **A**                    | Net product ($A=\sum a_{i}$)            |       |                                     |
| **B**                    | Depreciation                           |       |                                     |
|                          | **Supply of Goods and Services**        |       | **Demand for Goods and Services**   |

Where net income is the aggregate of wages, profits, rents, and interest. It is important to remember that depreciation is not a transaction of any kind, but rather the devaluation—or, more precisely, a loss of value—of the capital stock, recorded in accounting terms.

The debit side brings together the sources of the supply of goods and services available in the economy, while the credit side records the uses of that supply, that is, where goods and services are allocated: consumption, inventory investment, fixed capital formation, and exports. Thus, debit corresponds to the origin or supply of resources, while credit corresponds to their destination.

### Introducing the government

The government collects direct taxes (which are levied on income or wealth, such as property tax and vehicle tax) and indirect taxes (which are levied on prices, such as VAT and excise taxes). Transfers are direct taxes with a negative sign; subsidies are indirect taxes with a negative sign. The two most important categories of transfers are, on the one hand, pensions and retirement benefits and, on the other hand, interest on public debt.

Tax is the generic designation for any type of revenue that the government is able to collect by virtue of being the government. Taxes (direct and indirect) are the most well-known form of public revenue, but in addition to them there are also fees (such as waste collection fees), betterment contributions (arising from public works), and other types of contributions (such as social security contributions and economic intervention levies, among others).

As for subsidies, in most cases they do not represent a redistribution of revenue collected through taxes, but rather the government’s waiver of revenue to which it would otherwise be entitled. For example, the government may wish to reduce the price of milk for final consumers and therefore waive the collection of the value-added tax (VAT) that would normally be levied on milk sales.

To deal with changes in prices due to taxes and subsidies, two concepts of product were introduced: the product at market prices, which includes the value of indirect taxes net of subsidies, and the product at factor cost, which does not include this additional value.

Now, the economy’s aggregate supply in a given period is the sum of gross domestic product at market prices and imports of non-factor goods and services. Aggregate demand, on the other hand, is the sum of GDP at market prices and exports of non-factor goods and services. The production framework is then:


|         | **Debit**                               |       | **Credit**                          |
|---------|-----------------------------------------|-------|-------------------------------------|
| **I**   | Imports of goods and services           | **G** | Exports of goods and services       |
| **J-H** | Net income sent [+] abroad              | **C** | Personal consumption                |
| **A**   | Net income ($\sum a_i$)                 | **L** | Government consumption              |
| **B**   | Depreciation                            | **D** | Change in inventories               |
| **Q-N** | Net indirect taxes                      | **E** | Gross fixed capital formation       |
|         | **Supply of Goods and Services**        |       | **Demand for Goods and Services**   |


Net indirect taxes are indirect taxes ($Q$) minus subsidies ($N$). We note that the government now appears as an additional source of demand. However, we do not directly find transfers, direct taxes, the government's current account balance, or other net current revenues, since here we are looking only at the product account and therefore consider only those items that directly affect output. These other elements appear, for example, in the appropriation account:

|         | **Debit**                        |                    | **Credit**      |
|---------|----------------------------------|--------------------|-----------------|
| **C**   | Personal consumption             | $\boldsymbol{a}_{1}$ | wages           |
| **P-M** | Net direct taxes                 | $\boldsymbol{a}_{2}$ | profits         |
| **R**   | Other net current receipts       | $\boldsymbol{a}_{3}$ | rents           |
| **F**   | Net private saving               | $\boldsymbol{a}_{4}$ | interest        |
|         | **Allocation of Net Income**     | **A**              | **Net Income**  |

Net direct taxes are direct taxes ($P$) minus transfers ($M$). To understand net private saving, we now turn to the capital account:

|       | **Debit**                              |       | **Credit**                |
|-------|----------------------------------------|-------|---------------------------|
| **D** | Change in inventories                  | **F** | Net private saving        |
| **E** | Gross fixed capital formation          | **B** | Depreciation              |
| **K** | Balance of payments current account balance | **O** | Government current account balance |
|       | **Total Gross Investment**             |       | **Total Gross Saving**    |

If we were to return to a closed economy without government, the credit side would contain only $F$ and $B$. Thus, net private saving is the portion of production that was not consumed, net of depreciation; that is,
$$
F = D + E - B,
$$
where $B$ is depreciation, $D$ is the change in inventories, and $E$ is gross fixed capital formation.

Other accounts that I will not discuss in detail at this stage include the external sector account:

|       | **Debit**                              |   | **Credit**                    |
|-------|----------------------------------------|---|-------------------------------|
| **G** | Exports of goods and services          | **I** | Imports of goods and services |
| **H** | Income received from abroad            | **J** | Income sent abroad            |
| **K** | Balance of payments current account balance |   |                               |
|       | **Total Debit**                        |   | **Total Credit**              |

And the government account:

|       | **Debit**                          |       | **Credit**                      |
|-------|------------------------------------|-------|---------------------------------|
| **L** | Government consumption             | **P** | Direct taxes                    |
| **M** | Transfers                          | **Q** | Indirect taxes                  |
| **N** | Subsidies                          | **R** | Other net current receipts      |
| **O** | Government current account balance |       |                                 |
|       | **Use of Revenue**                 |       | **Total Revenue**               |

Direct taxes are levied on both firms ($P.1$) and households ($P.2$). Item $R$ must also be included because income recipients make, in addition to direct tax payments, other types of payments directly to the government (such as fees, social security contributions, fines, or rents for the use of government-owned property), which constitute the government's other current revenues.

We must also consider $O$, because once the government is introduced into the system, it also becomes a source of saving. Thus, it represents the third source of saving, alongside the private sector and the external sector.

The internal consistency of the accounts is assumed to hold. One way to verify this is to sum the debit side of all five accounts and subtract from it the sum of the credit side of all five accounts. If the system is indeed balanced, the result of this operation must be zero.

$$
[(I+J-H+A+B+Q-N)+(C+P-M+R+F)+(D+E)+(G+H+K+L+M+N+O)]
$$

$$
-
$$

$$
[(G+C+L+D+E)+(A)+(F+B+K+O)+(I+J)+(P+Q+R)]
$$

$$
=0
$$

It is important to emphasize that this is only one model of the System of National Accounts (SNA 93), and it is not the version currently in use. However, because it is considerably simpler, I found it more suitable for exploration. Moreover, I placed particular emphasis on the product account.

### Input--Output Matrix

The input--output matrix, whose development is associated with the Nobel Prize-winning economist Wassily W. Leontief (1906-1999), aims to provide an analysis of the intersectoral relationships in production and can be viewed as an alternative instrument to the system of national accounts.

From an input-output matrix, it is possible, for example, to estimate the impact of an increase or a contraction in the production of a given sector on the overall level of production, employment, and sectoral demand—a type of information that a conventional system of national accounts is not able to provide.

A very simple example may be useful for understanding the idea behind the input-output matrix, as well as how it operates and what makes it useful. Consider a hypothetical economy with only three sectors—say, 1, 2, and 3—that engage in economic transactions with one another. If $X_{ij}$ represents the sales from sector $i$ to sector $j$, we can construct the following matrix:
|                      |   |   | **Sectors** |        | **Final Demand** | **Output** |
|----------------------|---|--------|-------------|--------|------------------|------------|
|                      |   | 1      | 2           | 3      |                  |            |
|                      | 1 | $x_{11}$ | $x_{12}$  | $x_{13}$ | $y_{1}$        | $x_{1}$    |
| **Sectors**          | 2 | $x_{21}$ | $x_{22}$  | $x_{23}$ | $y_{2}$        | $x_{2}$    |
|                      | 3 | $x_{31}$ | $x_{32}$  | $x_{33}$ | $y_{3}$        | $x_{3}$    |
| **Value Added**      |   | $v_{1}$  | $v_{2}$   | $v_{3}$  |                |            |
| **Output**           |   | $x_{1}$  | $x_{2}$   | $x_{3}$  |                |            |

Considering the sales from sector $i$ to sector $j$ as a fraction of the total output of sector $j$, we have

$$
x_{ij}=a_{ij}x_{j}.
$$

We can then define a matrix of technical coefficients:

|   |      1      |      2      |      3      |
|:-:|:-----------:|:-----------:|:-----------:|
| 1 |    $a_{11}$    |    $a_{12}$    |    $a_{13}$    |
| 2 |    $a_{21}$    |    $a_{22}$    |    $a_{23}$    |
| 3 |    $a_{31}$    |    $a_{32}$    |    $a_{33}$    |

We can then write

$$
AX+Y=X,
$$

where $A$ is the matrix of technical coefficients, and $X$ ($Y$) is the column vector whose elements are $x_i$ ($y_i$). In other words, we may think of the problem from another perspective: what is the final demand for sector $i$? It is the net output of sector $i$, that is, $x_i$ minus the sales that sector $i$ must make to all other sectors:

$$
y_i=x_i-\left(\sum_j x_{ij}\right)
=x_i-\left(\sum_j a_{ij}x_j\right).
$$

Rearranging this expression, and considering the case of three sectors, we obtain

$$
x_i=a_{i1}x_1+a_{i2}x_2+a_{i3}x_3+y_i.
$$

Notice that $a_{ii}$ represents the "sales" of sector $i$ to itself. This can be interpreted as the fraction of sector $i$'s own production that is consumed internally as an input in its production process.

We can further rearrange the matrix formulation as

$$
AX+Y=X,
$$

$$
(I-A)X=Y,
$$

$$
X=(I-A)^{-1}Y,
$$

$$
X=LY,
$$

where $L$ is the Leontief matrix, computed from the matrix of technical coefficients. Given a fixed demand vector $Y$, this matrix allows us to calculate the level of production required in each sector to satisfy that demand.

We will now work through an example adapted from the one presented in the book.

In [ ]:
# @title
import numpy as np

In [ ]:
# @title
A = np.array([[45/500, 240/2900,15/1952],
              [90/500, 600/2900,210/1952],
              [0.00, 144/2900, 0.00]])
print('A matriz de coeficientes A é:')
print(A)
I = np.eye(3)
M=I-A
L = np.linalg.inv(M)
print('A matriz de Leontieff é:')
print(L)

A matriz de coeficientes A é:
[[0.09       0.08275862 0.00768443]
 [0.18       0.20689655 0.10758197]
 [0.         0.04965517 0.        ]]
A matriz L é:
[[1.12233089 0.11845059 0.02136762]
 [0.25644763 1.29648526 0.14144909]
 [0.01273395 0.0643772  1.00702368]]


We begin with the following matrix of technical coefficients:
| 1 |   2  |   3  |      |
|:-:|:----:|:----:|:----:|
| 1 | 0.09 | 0.08 | 0.01 |
| 2 | 0.18 | 0.21 | 0.11 |
| 3 | 0.00 | 0.05 | 0.00 |

We can now compute the Leontief matrix. From this point on, we will adopt vector notation:

$$
L=\left(I-A\right)^{-1}
=
\left[
\begin{array}{ccc}
1.12 & 0.11 & 0.02\\
0.26 & 1.30 & 0.15\\
0.01 & 0.06 & 1.01
\end{array}
\right].
$$

In [ ]:
# @title
v = np.array([[200],[2000],[1808]])
print('Se a demanda é:')
print(v)
x=np.dot(L, v)
print('Então a produção bruta necessária é:')
print(x)

Se a demanda é:
[[ 200]
 [2000]
 [1808]]
Então a produção bruta necessária é:
[[ 500.]
 [2900.]
 [1952.]]


If we have a demand vector

$$
y=
\left[
\begin{array}{ccc}
200 & 2000 & 1808
\end{array}
\right]^{T},
$$

then the required gross output is

$$
x=Ly=
\left[
\begin{array}{c}
500\\
2900\\
1952
\end{array}
\right].
$$

**Remark:** Although it is not explicitly stated, all quantities here should be interpreted as being measured in monetary values (prices).

In [ ]:
# @title
print('A produção Bruta total é:',sum(x)[0])
print('A demanda total é:',sum(v)[0])
X = np.array([[A[0][0]*x[0][0],A[0][1]*x[1][0],A[0][2]*x[2][0]],
              [A[1][0]*x[0][0],A[1][1]*x[1][0],A[1][2]*x[2][0]],
              [A[2][0]*x[0][0],A[2][1]*x[1][0],A[2][2]*x[2][0]]])
print('E a matriz de insumos-produtos é:')
print(X)
print('E o valor adicionado é:')
v = np.array([[x[0][0]-np.sum(X[:, 0])],[x[1][0]-np.sum(X[:, 1])],[x[2][0]-np.sum(X[:, 2])]])
print(v)

A produção Bruta total é: 5352.0
A demanda total é: 4007.9999999999995
E a matriz de insumos-produtos é:
[[ 45. 240.  15.]
 [ 90. 600. 210.]
 [  0. 144.   0.]]
E o valor adicionado é:
[[ 365.]
 [1916.]
 [1727.]]


Furthermore, since $x_{ij}=a_{ij}x_j,$
we can recover the transaction matrix $X$:

|                 |   |       | **Sectors** |       | **Final Demand** | **Output** |
|-----------------|---|-------|-------------|-------|------------------|------------|
|                 |   | 1     | 2           | 3     |                  |            |
|                 | 1 | 45    | 240         | 15    | 200              | 500        |
| **Sectors**     | 2 | 90    | 600         | 210   | 2000             | 2900       |
|                 | 3 | 0     | 144         | 0     | 1808             | 1952       |
| **Value Added** |   | 365 | 1916       | 1727 | 4008             |            |
| **Output**      |   | 500   | 2900        | 1952  |                  | 5352       |

The total output is simply the sum of the output of each sector, $\sum_i x_i.$ Similarly, total final demand is the sum of the final demand of each sector, $\sum_i y_i.$

It is an accounting identity that total value added is equal to total final demand, since total value added represents the value of the economy's surplus.

However, to calculate the value added of a given sector, we must consider its gross output, $x_i$, and subtract the inputs purchased from all sectors, $\sum_j x_{ji}.$

Therefore, the value added of sector $i$ is given by $v_i=x_i-\sum_j x_{ji}.$  For example, sector 1 produces 500, but it consumes 45 units from itself and 90 units from sector 2. Hence, its value added is

$$
500-90-45=365.
$$

Therefore, value added measures the contribution of each sector to income generation, whereas final demand measures the final destination of production. When aggregated over the entire economy, the two are equal.

## National Accounts and Macroeconomics

National accounting emerged alongside the development of Keynesian theory, when the English economist John Maynard Keynes, in the mid-1930s, wrote *The General Theory of Employment, Interest and Money* to challenge the prevailing economic theory and argue that the economy did not possess automatic mechanisms capable of escaping recessions and unemployment. It is worth highlighting the marginalist approach to unemployment. Under this approach, all existing unemployment was regarded as voluntary unemployment; that is, individuals who were not working were assumed to be in that situation because they were unwilling to supply their labor at the prevailing wage. In other words, they were unemployed because they chose not to work. The Great Depression of the 1930s clearly demonstrated the inadequacy of this theory in explaining reality. Keynes therefore sought to show that no such automatic regulator existed and that, consequently, most unemployment was involuntary, resulting from insufficient demand for labor and therefore an inability to employ the available labor supply. To demonstrate this, Keynes had to carry out a genuine revolution in economic thought, overturning several of the postulates that formed the backbone of the then-dominant theory.

For Keynes, the economic theory with which economists of the time worked was dominated, either implicitly or explicitly, by Say's Law, according to which supply creates its own demand. As a consequence, there was little concern with developing a theory capable of explaining the determinants of aggregate demand.

However, the "Keynesian consensus" was broken in the mid-1970s with the emergence of the rational expectations theory, which revived the assumptions that Keynes had challenged and restored the primacy of orthodox economic theory. The combination of inflation and unemployment that characterized the late 1970s led to widespread questioning of the role of the State as a regulator of effective demand, bringing to the forefront policies associated with what came to be known as neoliberalism (deregulation, control of public spending, a minimal state, and privatization).

It is important to emphasize that the system of national accounts was little, if at all, affected by this shift. Macroeconomic identities are not, by themselves, indicators of causal relationships among the variables they involve. Today, the identity between product, income, and expenditure is not questioned by anyone, regardless of whether one accepts the causal relationships proposed by Keynesian theory.

Let us begin by considering a closed economy without government. If we denote income by $Y$, consumption by $C$, and investment by $I$, then $Y=C+I$.

We now wish to deepen our understanding by asking, within Keynesian theory, what determines $C$ and $I$. According to Keynes, the principal determinant of consumption is income, $Y$. However, for a given increase in income, consumption increases by less than proportionally, since there exists what Keynes called the propensity to consume, which follows from what he termed the fundamental psychological law. In other words, for a given level of income, households consume most of it but also save a fraction. The propensity to consume is much higher among low-income households (in the limiting case, extremely low-income households save nothing and consume all of their income) than among high-income households. Thus, we may define a propensity to consume satisfying $0 < c < 1$, where $ C=f(Y)$.

Suppose now that there exists a component of consumption that does not depend on income, called autonomous consumption and denoted by $C_a$. Then, $C=C_a+cY,$ so that $Y=C_a+cY+I$.

Notice that if $I=0$, then whenever $C_a>0$, it is necessary that $c < 1$ for the equality $Y=C$ to hold. More explicitly, $C_a=(1-c)Y$.

Solving for $Y$, we obtain

$$
Y=\frac{C_a+I}{1-c}.
$$

As for investment, Keynes argued that it depends on two variables. These variables are the interest rate, $r$ (more precisely, the underlying variable is liquidity preference, that is, the desire of individuals and banks to hold money. According to Keynes, liquidity preference is the fundamental determinant of the economy's interest rate) , and the marginal efficiency of capital, $E_m$ (more precisely, this refers to expectations regarding the future returns on capital goods, that is, the expected future profitability of new investment. These expectations determine the marginal efficiency of capital). Thus, we may write $I=f\left(r,E_m\right)$.

Investment is therefore highly unstable. Since the Keynesian theory of the determinants of investment is extremely rich and complex, explaining it in detail would require several chapters, which is well beyond the scope of this text.

Returning to the expression for $Y$, Keynes called the quantity $\frac{1}{1-c}$ the **multiplier**. If we define $
\omega=\left(1-c\right)^{-1}$, then $Y=\omega\left(C_a+I\right)$.

Notice that the greater the propensity to consume—that is, the larger the value of $c$—the greater the multiplier $\omega$. If $C_a$ is relatively stable, then the behavior of $Y$ becomes increasingly dependent on the behavior of $I$, which is highly unstable.

Let us now consider an open economy with government. In addition to private consumption, $C$, and investment, $I$, we introduce government expenditure, $G$, exports, $X$, and imports, $M$. Income is then given by

$$
Y=C+I+G+\left(X-M\right).
$$

The aggregates measured from the **domestic** perspective correspond to the total value produced within the country's territory, regardless of the origin of the factors of production responsible for that output. In contrast, aggregates measured from the **national** perspective correspond to the value added generated by production factors owned by residents, regardless of the territory in which that value is generated.

Extending to this broader expression the same considerations previously made for a closed economy without government, we see that the level of income at which the economy operates depends not only on consumption and investment, but also on government expenditure and net exports. The same assumptions previously made regarding $C_a$ and $I$ also apply to these additional variables. Therefore,

$$
Y=\frac{1}{1-c}\left[C_a+I+G+\left(X-M\right)\right].
$$

A particularly intuitive way to understand this process is to think of a mechanism of stimuli and leakages that continuously influence the levels of income and output. If there is an increase in the autonomous component of consumption, in investment, in government expenditure, or in the external demand for the goods and services produced by the economy, any of these increases will stimulate production and raise the level of income by an amount determined by the multiplier.

In the case of exports, this constitutes an external stimulus, or, in other words, an injection of demand into the economy, resulting from an increase in foreign demand for domestically produced goods and services. Conversely, an increase in imports represents a leakage of stimulus, that is, a transfer of part of the economy's demand for goods and services to the rest of the world.

We may also appreciate the importance attributed to the government as a consequence of Keynes's analysis of the determinants of income. If an increase in the level of income and output at which the economy operates can result from higher government expenditure, then the government assumes an important role beyond those traditionally assigned to it.

One of the approaches to measuring output is precisely the demand (or expenditure) approach, which decomposes GDP into the categories of demand, as represented by the equation

$$
Y=C+I+G+\left(X-M\right).
$$

## Accounting Identities

The following identities derive from the basic identities product $\equiv$ expenditure $\equiv$ income and investment $\equiv$ savings. Before that, however, we must distinguish between aggregates measured from the:

• **Domestic perspective**: they measure the total value produced within the territory of the country, regardless of the origin of the factors responsible for that production.

• **National perspective**: they consider the value added generated by factors of production owned by residents, regardless of the territory where that value is generated.

Thus:

• **Domestic product**: reflects the total output produced within the country’s territory, regardless of the origin of the factors of production responsible for it.

• **National income**: considers the value added generated by factors of production owned by residents, regardless of the territory where that value is generated.

We define:

• GNI = Gross National Income  
• GDP = Gross Domestic Product  
• NFI = Net Factor Income from Abroad
Then we have:

$$
\text{GNI} = \text{GDP} - \text{NFI}.
$$

Now, if $TR$ denotes net unilateral transfers received, then:

$$
\text{GNDI} = \text{GNI} + TR,
$$

where GNDI is Gross National Disposable Income. From this, we can rewrite:

$$
\text{GDP} = \text{GNDI} + \text{NFI} - TR.
$$

Considering the government explicitly, we define **government net income (RLG)** as the sum of direct taxes, indirect taxes, and other current government revenues, minus transfers. We can then rewrite:

$$
\text{GNDI} = RPD + RLG,
$$

where $RPD$ is private disposable income.

Now, considering both the nature of demand and the allocation of income, let us briefly return to the idea of a closed economy without government. In this case, on the one hand, output ($Y$) equals consumption demand ($C$) and investment demand ($I$), while on the other hand income ($Y$) equals consumption ($C$) plus savings ($S$). Formally, $Y = C + I$ and $ Y = C + S,$ so that $I = S$.

Introducing the government, we obtain an additional demand component through government expenditure $G$, so that $Y = C + I + G$, and also an additional income component through government net income $RLG$, so that $Y = C + S + RLG$ or equivalently,

$$
I = S + RLG - G = S + S_g = S_D,
$$

where $S_g = RLG - G$ is government saving, and $S$ is now explicitly private saving. We also denote $S_D = S + S_g$ as domestic saving, i.e., total gross saving of the economy.

Reintroducing the external sector, we obtain:

$$
I = S + S_g + S_e,
$$

where external saving is defined as

$$
S_e = (M - X) +  NFI - TR.
$$

We can show this by starting from the expenditure-side identity:

$$
Y = C + I + G + (X - M) =  GDP,
$$

and from the income-side identity:

$$
Y = C + S + RLG +  NFI - TR = GNDI  +  NFI - TR =  GDP,
$$

when viewed from the allocation of income perspective.Note that $GNDI  = C + S + RLG$, i.e., Gross Disposable Income is the sum of private consumption, private saving, and government net income. Since $GNDI  = RPD + RLG$, it follows that $RPD = C + S$, i.e., private disposable income is the sum of consumption and savings.

Thus,

$$
I + G + (X - M) = S + RLG +  NFI - TR,
$$

$$
I = S + [RLG - G] + [(M - X) +  NFI - TR],
$$

$$
I = S + S_g + S_e.
$$

It is worth remembering that all these results are based on the SNA 93 framework, which is not the most recent version. However, it is used here for pedagogical purposes, as it helps clarify several underlying mathematical relationships and may also be useful for future modeling exercises.